### DISCUSSION TITLE
会拖慢保存文件的速度和其他插件的执行速度

### DISCUSSION THREAD

**Discussion by nookery** [codingyu/laravel-goto-view#52] [General]
一个300多行的blade文件，随意修改再保存，基本是瞬间保存；
安装这个插件，要大概等2秒才能保存成功。

另外，如果还有其他插件，其他插件的执行速度也变得特别慢，比如这个插件`Laravel Blade Spacer`，和`Laravel goto view`共同使用的话，要3秒才能起作用。

复现步骤：

1. 用vscode打开一个300多行的blade文件
1. 然后在Remote-Container中打开
2. 然后测试上面的问题

**Comment by nookery** [comment]
#48 
可能和这个是一个问题。

**Comment by codingyu** [comment]
嗯，这个扩展本质是根据字符串（匹配相关命名）去判断文件是否存在， 
我还没找到什么好的方式重新实现（比如借助php，或者解析单个PHP文件）
Remote-Container 是不是本身还会有 io 速度的问题？

**Comment by nookery** [comment]
> 嗯，这个扩展本质是根据字符串（匹配相关命名）去判断文件是否存在，
> 我还没找到什么好的方式重新实现（比如借助php，或者解析单个PHP文件）
> Remote-Container 是不是本身还会有 io 速度的问题？

宿主机文件行数多也没发现慢的问题，Remote-Container 文件行数少的话不明显，多就慢的明显，Remote-Container 性能肯定是差一些的。

**Comment by nookery** [comment]
看来是Docker的问题，如果是Mac，开启这个选项，能改善很多。
<img width="1174" alt="截屏2022-02-20 下午1 23 57" src="https://user-images.githubusercontent.com/5194193/154829691-f41e052b-262f-4160-93a0-31e59412c390.png">
